### DISCUSSION TITLE
Anyone having problems importing FCP XMLs into other apps?

### DISCUSSION THREAD

**Discussion by tangierc** [latenitefilms/BRAWToolbox#134] [General]
I am just curious if anyone is having any issues using BRAW Toolbox-generated clips and third party apps such as X2Pro or even Apple Apple such as Logic Pro X.

I have iteratively deleted problem files in my FCP-exported XML so that I can use X2Pro to generate an AAF. The problems always show up when attempting to reimport the XML into FCP.
I have cleaned up every problem except one warning I can't figure out which says: 

"Invalid edit with no respective media. (/fcpxml[1]/library[1]/event[1]/project[1]/sequence[1]/spine[1]/asset-clip[17])"

I don't believe it's a problem with BRAW Toolbox, but since I still can't get X2Pro to parse the XML, I tried Logic Pro to make the AAF instead and I am getting a large amount of errors when trying to import the XML which leads me to believe that there are other issues than just the one above; perhaps clips it can't make sense of. This error is the same and repeated:

The operation couldn’t be completed. (OSStatus error -50.)
I can't figure out what specific error this is since there are several options on this page:
https://www.osstatus.com/search/results?platform=all&framework=all&search=-50

I am dead in the water getting this large project out of FCP to create an AAF using X2Pro or Logic Pro.
If you have any suggestions, I could use them.

![Screenshot 2023-06-09 at 6 36 45 PM](https://github.com/latenitefilms/BRAWToolbox/assets/9577806/926fc3b3-08f8-4897-bedd-1619d6811953)

![Screenshot 2023-06-08 at 7 38 52 PM](https://github.com/latenitefilms/BRAWToolbox/assets/9577806/45413d1b-c5f4-4405-98ab-ba3e7d04d81a)

[Anxious-TC Cut 5-Locked Picture for Teams-Problem Audio Removed.fcpxml.zip](https://github.com/latenitefilms/BRAWToolbox/files/11711217/Anxious-TC.Cut.5-Locked.Picture.for.Teams-Problem.Audio.Removed.fcpxml.zip)

**Comment by latenitefilms** [comment]
Have you tried using the [Convert PROJECT to Resolve Friendly FCPXML](https://brawtoolbox.io/toolbox/#convert-project-to-resolve-friendly-fcpxml) Toolbox to generate the FCPXML that you send to X2Pro?

I assume the reason you're seeing that error is that BRAW Toolbox clips don't actually contain any "media" - they're just a Generator with an Effect applied.

If using the Toolbox still doesn't work, you can try opening it up in a text editor and do a find and replace to remove the `.braw` extension and replace it with a `.mov` extension - just incase X2Pro can't handle BRAW files, etc.

**Reply by latenitefilms** [reply]
Also, you can try just unlinking all the audio, then delete all the video, as your FCPXML only needs audio if you're sending it to X2Pro. That will remove BRAW Toolbox completely out of the equation.

**Reply by tangierc** [reply]
That was going to be my next step after trying to resolve this last warning. I wanted to try and clean up the XML as much as I could once I before going that route so that I'd be sure I did everything to get FCP to reimport it's own XML without issue.

Thanks for that advice of replacing the .braw. I'll try that with BBedit.  

Support on X2pro told me that the invalid edit warning I am getting is at a place in the timeline where I don't actually have the supposed problem audio clip so that didn't work.  I deleted all instance of that particular audio clip on the timeline and tried exporting/reimporting the XML and it still gave the same error.

**Reply by latenitefilms** [reply]
BRAW Toolbox just uses regular AIFF files for audio, so after you delete all the video assets, you should only have AIFF files, so hopefully that'll make problem solving with the X2Pro developers a lot easier.

**Reply by tangierc** [reply]
I made a duplicate timeline with all of the video deleted. Then I used Convert PROJECT to Resolve Friendly FCPXML.
There were far fewer errors in X2Pro (see image), but Logic Pro still gives the OSStatus error -50) over and over again on import.

I kept working on the errors that X2Pro would give. There were only three (see image below), but every time I exported a new XML using the Toolbox a new braw clip came up in the X2Pro error window. I found out that it keep being multicam clips with BRAW Toolbox clips in it. I removed all of those and fixed all subsequent issues to the point where using the Convert Project to Resolve Friendly XML Toolbox told me that there weren't any BRAW Toolbox clips detected. So I thought that was a good thing since now I'm at just the audio on the timeline.

I exported an XML from FCP (not using Toolbox this time) and got more errors than I ever got using the Toolbox.
This is so puzzling. I've completed many films with this workflow, but have never had this much difficulty getting my FCP XML to work to create an AAF and no additional path to do so. I'm not sure where to go from here. Logic pro still throws out that same error over and over. Here's the latest XML if you want to have a look. I'm going to send it to X2Pro support as well, but I don't know if they can do much more. I sent feedback to Apple on all fronts for FCP and Logic.

I can't help but wonder if the toolbox still allowed me to export a Resolve Friendly XML that it may pass X2Pro. The Resolve friendly version always had fewer errors, but now that I have no braw clips in my timeline, I can't use that Toolbox feature, must use FCP's export function which unfortunately yields many more errors.

[XMLs.zip](https://github.com/latenitefilms/BRAWToolbox/files/11711461/XMLs.zip)


<img width="812" alt="Screen Shot 2023-06-09 at 8 04 00 PM" src="https://github.com/latenitefilms/BRAWToolbox/assets/9577806/d34148f2-0d1d-4680-91c9-1a75d54ba22a">



<img width="812" alt="Screen Shot 2023-06-09 at 8 55 42 PM" src="https://github.com/latenitefilms/BRAWToolbox/assets/9577806/656bb368-4e52-4c1a-ba86-53dc32986ca5">

**Reply by latenitefilms** [reply]
I will have a look tonight when back in front of my laptop. I think the issue is that X2Pro never considered a scenario where Multicam Clips would contain assets with no actual video tracks.

One easy workaround would just be to send a FCPXML to Resolve, and export the AAF from Resolve. Everything should go from FCPX to Resolve perfectly with the Toolbox.

**Reply by tangierc** [reply]
I went the DaVinci Resolve route after using the Convert Project to Resolve Friendly FCPXML and managed to get an AAF out. I lose my role-based organization that I get with X2Pro, but at least this worked. The only oddity was that every braw clip shows up black compound clips in Resolve (18). Only the Red footage actually showed up viewable in Resolve. Will revisit instructions Toolbox instructions to see if there's something I didn't do.

**Comment by tangierc** [comment]
That may be the only way to go. I need to step away from my computer for a bit, but will try that next.  I updated my previous post at the end, regarding your Convert to Resolve Friendly Toolbox yielding the least errors in X2Pro. However now that I have no braw clips in the timeline I can’t get the benefit of exporting an XML using the Toolbox.

Thanks

**Comment by latenitefilms** [comment]
Just to recap everything so there's no confusion or ambiguity...

---

In the current version of **BRAW Toolbox**, we default to creating Multicam Clips to get around [this bug](https://github.com/latenitefilms/BRAWToolbox/issues/128). However, earlier versions of **BRAW Toolbox** created Synchronised Clips by default, which is what you've done for your feature film.

To clarify... there's nothing wrong with using Synchronised Clips - they SHOULD work - it's a Final Cut Pro bug/limitation/oversight that is causing the glitches. Essentially BRAW Toolbox is just using Generators in a way that was never originally envisaged.

---

In **BRAW Toolbox**, when importing BRAW clips as Synchronised Clips they'll appear in the timeline like this:

<img width="280" alt="image" src="https://github.com/latenitefilms/BRAWToolbox/assets/22286696/dbb9c8cb-b1da-41f1-9cf9-5ddfcd0a9673">

If you double click on the Synchronised Clip in the timeline (or select **Clip > Open Clip** from the menubar) you'll see this:

<img width="1060" alt="image" src="https://github.com/latenitefilms/BRAWToolbox/assets/22286696/1586539a-72ff-46c7-92f7-86a5dde09ea3">

The Black Video layer is just a stock-standard **Custom** Solids Generator:

<img width="529" alt="image" src="https://github.com/latenitefilms/BRAWToolbox/assets/22286696/3b140641-afdf-4614-a051-856188e6fbe4">

By itself, this **Custom** Generator doesn't do anything - it just outputs Black.

---

> As a side-note, annoyingly, as discussed [here](https://github.com/latenitefilms/BRAWToolbox/issues/128#issuecomment-1579617836) this **Custom** Generator is actually the reason we have issues with Synchronised Clips. Because the Custom Generator is normally a static image, Final Cut Pro is (annoyingly) incorrectly assuming the image hasn't changed, so it's triggering a render - which is why you get glitches. Adding keyframes to the Color Parameter actually fixes things. I will push out an update this week to finally solve this and add some kind of toolbox to fix older projects like yours.
> 
> <img width="501" alt="image" src="https://github.com/latenitefilms/BRAWToolbox/assets/22286696/b3e082ed-497d-4cd1-bac8-8cd4b02da8b2">

---

The "magic" that **BRAW Toolbox** adds is the **BRAW Toolbox** effect that we apply to the **Custom** generator. This is what renders the `BRAW` image in Final Cut Pro.

<img width="495" alt="image" src="https://github.com/latenitefilms/BRAWToolbox/assets/22286696/e37ad648-3d80-4fde-a1d1-25ebf71a1914">

The audio clip is just a regular `WAV` file - nothing fancy - when you import a clip from **BRAW Toolbox**, we "extract" the audio from the `BRAW` file and convert it to a `WAV` file.

<img width="1004" alt="image" src="https://github.com/latenitefilms/BRAWToolbox/assets/22286696/fe503f5c-23ad-43d8-ac2a-df1a6f25c476">

---

So in your case, you have **Synchronised Clips** in your timeline, and inside those Synchronised Clips is a **Custom** generator with the **BRAW Toolbox** effect applied (to render your `BRAW` images) and a regular `WAV` file.

If I export this single **BRAW Toolbox** clip timeline, it works fine out-of-the-box with **X2Pro**:

<img width="1397" alt="image" src="https://github.com/latenitefilms/BRAWToolbox/assets/22286696/f925bacb-864a-4a5f-b15d-84d3f97b18ec">

This AAF imports into **Logic Pro** without issue:

<img width="1317" alt="image" src="https://github.com/latenitefilms/BRAWToolbox/assets/22286696/518eae66-493b-49e2-95aa-d6f6dcd712ff">

---

However, if I use the **Convert PROJECT to Resolve Friendly FCPXML** to create a new `FCPXML` that's ready for Resolve, it won't work in **X2Pro**, because **X2Pro** doesn't support `BRAW` files - you'll see the same error you have been seeing:

<img width="812" alt="image" src="https://github.com/latenitefilms/BRAWToolbox/assets/22286696/e8cff401-229e-45bd-aa28-5b42767810ed">

I can however send this `FCPXML` to **DaVinci Resolve**, and it works fine, and you can export an `AAF` from there:

<img width="648" alt="image" src="https://github.com/latenitefilms/BRAWToolbox/assets/22286696/766859d5-5a77-45ee-a266-429be8fcfa7b">

---

Synchronised Clips can be broken apart using the **Clip > Break Apart Clip Items** menubar item (or **COMMAND+SHIFT+G**).

For example, before:

<img width="343" alt="image" src="https://github.com/latenitefilms/BRAWToolbox/assets/22286696/c40f2ab3-a691-439e-9409-7777a793c0a4">

...and after:

<img width="451" alt="image" src="https://github.com/latenitefilms/BRAWToolbox/assets/22286696/dd450775-5296-4d19-af87-c7da308d4631">

Now, because of the way Final Cut Pro works, after breaking apart, the **Custom** generator will most likely just display the wrong frames - which is why we have to "embed" it in a Synchronised Clip in the first place. However the audio will now just be a regular `WAV` on the timeline - exactly the same as if you dragged a `WAV` file from Finder to the timeline.

If I export this broken apart BRAW Toolbox clip via **X2Pro**, it also works fine:

<img width="1399" alt="image" src="https://github.com/latenitefilms/BRAWToolbox/assets/22286696/74964164-892f-4057-8149-36bdec544a5e">

This AAF also works fine with Logic Pro:

<img width="1287" alt="image" src="https://github.com/latenitefilms/BRAWToolbox/assets/22286696/472839b7-0698-4541-a46a-bae58aaefe52">

---

So, to summarise... out of the box, **BRAW Toolbox** should work without issues with **X2Pro** - simply because **BRAW Toolbox** is kinda unrelated, as we're just dealing with `WAV` files.

---

You should only use the **Convert PROJECT to Resolve Friendly FCPXML** feature if you're sending a FCPXML to DaVinci Resolve, or another app that supports `FCPXML` and `BRAW` files (like Autodesk Flame). Using this Toolbox won't work with **X2Pro**, because the `FCPXML` now references actual `BRAW` files, which **X2Pro** can't decode. 

---

My original suggestion [above](https://github.com/latenitefilms/BRAWToolbox/discussions/134#discussioncomment-6137882) was to open the `FCPXML` in a text editor and replace the `.braw` with `.mov`:

<img width="1311" alt="image" src="https://github.com/latenitefilms/BRAWToolbox/assets/22286696/374664b7-3d1f-4802-860d-5e73fdbfe8c0">

However, in actually testing, this doesn't work:

<img width="812" alt="image" src="https://github.com/latenitefilms/BRAWToolbox/assets/22286696/75e297fa-657e-4cab-9aed-da6ce15126fd">

This is because the `MOV` file doesn't actually exist, so **X2Pro** can't extract the audio from it - which makes sense.

---

So as you originally summarised, I don't think the original issues you were having were actually to do with **BRAW Toolbox** clips, as **BRAW Toolbox** clips should work fine with **X2Pro**, because again, they're just regular `WAV` files - nothing fancy, and **X2Pro** will just ignore the Generator anyway (as it has nothing to do with audio).

Unfortunately I can't really test this myself with your `FCPXML`, because I don't have any of the media, and **X2Pro** won't export anything unless it can read the media.

However, my suggestion would be to remove all your video files, and **Break Apart** all your **BRAW Toolbox** clips, and then you literally just have audio files in a timeline. If you're still having issues with **X2Pro**, then this would either be a bug in Final Cut Pro's output or the way X2Pro is handling things (which I assume they should be able to fix or provide advice on pretty quickly & easily). You can send them this thread to explain everything too.

<img width="1437" alt="image" src="https://github.com/latenitefilms/BRAWToolbox/assets/22286696/ef0104c6-108a-44d1-a1a7-ab8d7ac8763e">

---

In regards to the `OSStatus` error you're seeing in **Logic Pro**, `OSStatus` error codes are used to communicate the success or failure of an operation. `OSStatus` **error -50** represents a parameter error, typically indicating that an invalid parameter was passed to a function.

Here's the relevant entry from `MacErrors.h`:

```
paramErr                   = -50, /*error in user parameter list*/
```

In practical terms, if you're seeing this error, it means that Logic Pro has called a function or method with an argument that the function doesn't know how to handle. This might be because the argument is of the wrong type, it's out of an expected range, or it's an otherwise correct value that doesn't make sense in the current context.

What exactly is going wrong is basically impossible to determine without more information. You'd probably need to contact Apple to get help with this. My GUESS is that maybe there's something in the `FCPXML` that they were simply not expecting. However, in your case, if you can "break apart" everything, and only have audio in your timeline, then, in theory, there should be nothing left to trip Logic Pro up.

Hope this helps!

**Comment by tangierc** [comment]
Thanks for all of this Chris. I hope I didn’t say anything that led to this. I completely am with you all the way on this and understand the way BRAW Toolbox is working using an effect on a generator. The mystery is why none of my BRAW Toolbox synchronized clips did not appear in Resolve as visible video, but only black clips. I haven’t used them in FCP in any special way other than intended inside FCP and using BRAW Toolbox. This is after making a Resolve friendly XML. 

I don’t know if I’ll solve this one. I managed to get an AAF out of Resolve. In the process of doing these tests, I passed an audio-only XML from FCP to Resolve and one with my locked picture and that’s when I noticed only the Red footage appeared as viewable. The braw clips were in the timeline in their correct positions and duration, but just black.

Thanks for all of you support.

Sincerely,

Tangier

**Reply by latenitefilms** [reply]
When you sent the `FCPXML` to **DaVinci Resolve**, was this **BEFORE** you did all the timeline cleanup? Based on the above threads, it sounds like maybe you used the **Convert PROJECT to Resolve Friendly FCPXML** Toolbox after attempting to clean up the timeline?

Assuming you still have your original picture-locked edit, I'd suggest just trying to drag that project into the **Convert PROJECT to Resolve Friendly FCPXML** Toolbox, and send to **DaVinci Resolve**.

For example, here's a simple timeline with just **BRAW Toolbox** Synchronised Clips:

<img width="1420" alt="image" src="https://github.com/latenitefilms/BRAWToolbox/assets/22286696/b87dc3dd-5037-4217-9697-7d464e1382cd">

...and drag into the Toolbox, select a save location, and you'll get this success message:

<img width="982" alt="image" src="https://github.com/latenitefilms/BRAWToolbox/assets/22286696/60dcbcb5-7bd5-4f19-b688-cacca140eded">

I'll use these import settings in **DaVinci Resolve**:

<img width="791" alt="image" src="https://github.com/latenitefilms/BRAWToolbox/assets/22286696/25795ff1-b159-4f6a-b498-4a20ae3dffac">

This is what it looks like:

<img width="1257" alt="image" src="https://github.com/latenitefilms/BRAWToolbox/assets/22286696/1a6f2ef0-f9f2-4006-809d-87f9946f437f">

However, I can select all the clips then right click and select **Decompose/Flatten in Place**:

<img width="381" alt="image" src="https://github.com/latenitefilms/BRAWToolbox/assets/22286696/cfde8566-800b-40a5-ad0f-94eac107d95d">

Now they all look like this:

<img width="1547" alt="image" src="https://github.com/latenitefilms/BRAWToolbox/assets/22286696/b55db8e9-1c98-4151-8ffc-c477102a804b">

I can then go into the **Deliver** page and select the **Pro Tools** preset with **Export Video** unticked.

<img width="460" alt="image" src="https://github.com/latenitefilms/BRAWToolbox/assets/22286696/d30147a6-33b8-4bbf-9e6e-1ca13560f60a">

This imports into Logic Pro as expected:

<img width="1336" alt="image" src="https://github.com/latenitefilms/BRAWToolbox/assets/22286696/2acccb40-3554-404c-889a-d1d9eaae8df1">

**Comment by tangierc** [comment]
Thanks Chris. All is well The XML I went back to pertaining to my previous post was my original locked picture for teams; the only difference between the locked picture and the one for teams being the latter has a slate, countdown with 2-pop, and tail pop at the end.

I restarted my machine, exported an FCPXML directly out of FCP and another using the BRAW Toolbox Convert PROJECT to Resolve Friendly FCPXML feature and only the one from BRAW Toolbox worked in Resolve with the braw video clips visible. 

I can only surmise that it had to be my own human error in testing various iterations (with and without video or the problem m4a files) of FCPX XML files.  I didn't need the video at all, but since Resolve was playing nice enough to support an AAF export with the XMLs even with the problematic clips I tried to clean out before, it made sense to stick with the locked picture version (no deleted clips) in Resolve to insure I had everything.
